src/components/inputs/money-input/money-input.js

```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
         const fakeEvent = {
           persist: () => {},
           target: {
-            id: MoneyInput.getAmountInputId(this.props.id),
-            name: getAmountInputName(this.props.name),
+            id: MoneyInput.getAmountInputId(props.id),
+            name: getAmountInputName(props.name),
             value: formattedAmount,
           },
         };
-        this.props.onChange(fakeEvent);
+        props.onChange(fakeEvent);
       }
     }
-  };
+  }, [props.id, props.name, props.value.amount]);
```

Comment:
should we include the `props.value.currencyCode`?

Reply:
Yes we should include all the props used within the hook.

PS: do we have the eslint rules for hooks enabled?